imqsauth/net_tools.go

```diff
@@ -78,5 +80,11 @@ func getIPAdress(r *http.Request) string {
 			return ip
 		}
 	}
-	return ""
+	return r.RemoteAddr
+}
+
+func auditUserLogAction(auditor authaus.Auditor, username, clientIP, action string, actionType authaus.AuditActionType) {
```

Comment:
Instead of sending the clientIP to this function, rather send the http.Request object. Then, make this function extract the clientIP.
In addition, this function must extract the identity of the caller. The identity of the caller is more important than the IP.